src/main/java/uk/gov/companieshouse/api/accounts/service/impl/PreviousPeriodService.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.companieshouse.api.accounts.service.impl;
+
+import com.mongodb.DuplicateKeyException;
+import com.mongodb.MongoException;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.companieshouse.GenerateEtagUtil;
+import uk.gov.companieshouse.api.accounts.CompanyAccountsApplication;
+import uk.gov.companieshouse.api.accounts.Kind;
+import uk.gov.companieshouse.api.accounts.LinkType;
+import uk.gov.companieshouse.api.accounts.ResourceName;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.model.entity.PreviousPeriodEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.PreviousPeriod;
+import uk.gov.companieshouse.api.accounts.repository.PreviousPeriodRespository;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.service.Service;
```

Comment:
Ideally we'd like to rename this Service to `ResourceService` to avoid the explicit package definition of the `Service` Spring stereotype below. @jsimons1989, thoughts?

Reply:
Sounds good to rename to avoid the conflict. As long as the name satisfies the "is a" relationship with the implementations (e.g. PreviousPeriodService is a ResourceService) then it will be fine. The name of the interface is the contract the implementations need to satisfy.